### ISSUE TITLE
calc::globalInit() must handle case where paths from PATH don't exist

### ISSUE THREAD

**Issue by kordejong** [pcraster/pcraster#134]
This code calls into dev::prefix -> dev::pathToExecutable. This code throws a boost::filesystem_error in case a directory path does not exist. This crashes linkout applications.

Example code attached. Compile code and run like this (assuming all required shared libs accessible):

```bash
PATH=/root/bin:$PATH ~/tmp/link/pcrastertest2 ~/tmp/link/pcrasterFile.xml
```

**Comment by kordejong** [comment]
[link.tar.gz](https://github.com/pcraster/pcraster/files/793748/link.tar.gz)

**Comment by OliverSchmitz** [comment]
may also effect other applications
$ pcrcalc
ERROR: programming error uncaught exception (in main())
(on cartesius)

**Comment by seb45tian** [comment]
Has this been handled yet? I ran into the same issue when trying to use PCRaster in an HPC environment (self compiled or via Miniconda!) running on Centos 7.7. `pcrcalc` throws the error mentioned by @OliverSchmitz and when importing into python you'll get

    >>> import pcraster as pcr
    terminate called after throwing an instance of 'boost::filesystem::filesystem_error'
      what():  boost::filesystem::status: Permission denied: "/root/bin/PCRasterModelEngine"
    Aborted (core dumped)

$PATH has non-existent locations in it, as described by OP (e.g. /root/bin). Getting rid of those solves the problem; but this is kind of a major issue, isn't it?

**Comment by OliverSchmitz** [comment]
not yet, I'll bump the priority of this issue

**Comment by OliverSchmitz** [comment]
This should be fixed in the 4.3 branch, please try that one when compiling yourself.
